types/roslib/index.d.ts

```diff
@@ -29,10 +31,16 @@ declare namespace ROSLIB {
 		 *
 		 * @constructor
 		 * @param options - possible keys include:
-		 *   * url (optional) - the WebSocket URL for rosbridge (can be specified later with `connect`)
+		 *   * url (optional) - (can be specified later with `connect`) the WebSocket URL for rosbridge or the node server url to connect using socket.io (if socket.io exists in the page) <br>
+		 *   * groovyCompatibility - don't use interfaces that changed after the last groovy release or rosbridge_suite and related tools (defaults to true)
+		 *   * transportLibrary (optional) - one of 'websocket', 'workersocket' (default), 'socket.io' or RTCPeerConnection instance controlling how the connection is created in `connect`.
+		 *   * transportOptions (optional) - the options to use use when creating a connection. Currently only used if `transportLibrary` is RTCPeerConnection.
 		 */
 		constructor(options:{
-			url?: string
+			url?: string,
+			groovyCompatibility?: boolean,
+			transportLibrary?: 'websocket' | 'wokersocket' | 'socket.io' | RTCPeerConnection,
```

Comment:
`workersocket` is not listed as an option in the [original docstring](https://github.com/RobotWebTools/roslibjs/blob/1.0.1/src/core/Ros.js#L29)

```suggestion
			transportLibrary?: 'websocket' | 'socket.io' | RTCPeerConnection,
```